cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/GoalUtils.java

```diff
@@ -530,7 +530,10 @@ public static double computeResourceUtilizationBalanceThreshold(double avgUtiliz
 
     } else {
       if (isLowUtilization) {
-        return balancingConstraint.lowUtilizationThreshold(resource) * balanceMargin;
+        double resourceBalanceUtilizationUpperLimit =
+            Math.max(balancingConstraint.resourceBalancePercentage(resource) * avgUtilizationPercentage,
+                     balancingConstraint.lowUtilizationThreshold(resource));
+        return resourceBalanceUtilizationUpperLimit * balanceMargin;
```

Comment:
This looks incorrect to me.
Shouldn't we pick the maximum between the following:
1. `balancingConstraint.lowUtilizationThreshold(resource) * balanceMargin` and 
2. `avgUtilizationPercentage * (1 + balancePercentageWithMargin)`

Reply:
Right. This way does make more sense